storm-client/src/jvm/org/apache/storm/messaging/netty/Client.java

```diff
@@ -420,15 +432,15 @@ public void close() {
 
     private void waitForPendingMessagesToBeSent() {
         LOG.info("waiting up to {} ms to send {} pending messages to {}",
-                 PENDING_MESSAGES_FLUSH_TIMEOUT_MS, pendingMessages.get(), dstAddressPrefixedName);
-        long totalPendingMsgs = pendingMessages.get();
+                 PENDING_MESSAGES_FLUSH_TIMEOUT_MS, pendingMessages.getCount(), dstAddressPrefixedName);
+        long totalPendingMsgs = pendingMessages.getCount();
         long startMs = System.currentTimeMillis();
-        while (pendingMessages.get() != 0) {
+        while (pendingMessages.getCount() != 0) {
```

Comment:
I am worried about performance since meter.getCount() doesn't seem trivial. Maybe we should keep those AtomIntegers and add meters separately. They are essentially different things

Reply:
sounds reasonable.  Let me know when you finish the rest of the review and I'll update this.